packages/server/src/adapters/content-handlers/selectContentHandlerOrUnsupportedMediaType.ts

```diff
@@ -1,20 +1,35 @@
 import { TRPCError } from '../../@trpc/server';
-import type { BaseContentTypeHandler } from '../../@trpc/server/http';
+import type {
+  BaseContentTypeHandler,
+  HTTPHeaders,
+} from '../../@trpc/server/http';
+
+interface MinimalHandlerOpts {
+  req: {
+    headers: HTTPHeaders | Headers;
```

Comment:
This could probably only take `Headers` and do the conversion higher up, but that's a bigger refactor for another day

Reply:
Yeah that's fine, limited number of these types right now so we can wrangle it here